Change GreenhouseGas individuals to classes

akleinau opened the pull request:
closes #56.
Adds classes:
**PortionOfCarbonDioxide** (CO2): Carbon dioxide is a portion of matter with the chemical formula CO2. It is a gas that occurs naturally in the atmosphere as a trace gas and it is a greenhouse gas.
**PortionOfMethane** (CH4): Methane is a portion of matter with the chemical formular CH4. It is a gas that occurs naturally in the atmosphere as a trace gas and it is a greenhouse gas. As it can be oxidised it is a fuel.
**PortionOfNitrousOxide** (N2O): Nitrous oxide is a portion of matter with the chemical formula N2O. It is a gas that occurs naturally in the atmosphere as a trace gas and it is a greenhouse gas.
**PortionOfSulphurHexafluoride** (SF6): Sulphur hexafluoride is a portion of matter with the chemical formula SF6. It is a potent greenhouse gas from anthropogenic origin.
**PortionOfNitrogenTrifluoride** (NF3): Nitrogen trifluoride is a portion of matter with the chemical formula NF3. It is a potent greenhouse gas from anthropogenic origin.

Adds individual:
**antropogenic** (individual of type origin):"x is anthropogenic if it was created by human activity."?

Deletes individuals:
- CarbonDioxide, CO2
- Methane, CH4
- NitrousOxide, N2O
- SulphurHexafluoride, SF6
- NitrogenTrifluoride, NF3

I additionally saw that Hydrofluorocarbon and Perfluorocarbon were direct subclasses of FluorinatedGreenhouseGas instead of PortionOfMater so I changed them like I changed the classes above, they already had defs.
- Delete HFC and PFC and make those abbreviations alternative terms of the Hydrofluorocarbon and Perfluorocarbon
- add prefix PortionOf to Hydrofluorocarbon and Perfluorocarbon and move them to PortionOfMatter
- delete duplicate individuals of them

l-emele commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -269,6 +272,61 @@ Class: <http://openenergy-platform.org/ontology/oeo-physical/ChemicalEnergy>
+    SubClassOf: 
+        PortionOfMatter,
+        <http://purl.obolibrary.org/obo/RO_0000091> some GreenhouseGas
+    
```
Methane is also a fuel, this link is missing.

stap-m replied:
We should discuss #227 first before changing this.

akleinau replied:
It is PortionOfMatter and through equivalent classes now also a PortionOfFuel since I added the disposition Fuel

l-emele commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -269,6 +272,61 @@ Class: <http://openenergy-platform.org/ontology/oeo-physical/ChemicalEnergy>
+    SubClassOf: 
+        PortionOfMatter,
+        <http://purl.obolibrary.org/obo/RO_0000091> some GreenhouseGas
+    
```
PortionOfNitrogenTrifluoride should not be treated differently to PortionOfHydrofluorocarbon, PortionOfPerfluorocarbon and PortionOfSulphurHexafluoride. These are subclasses of:
* 'has disposition' some FluorinatedGreenhouseGas
* PortionOfMatter

akleinau replied:
I implemented this now

stap-m commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -269,6 +272,61 @@ Class: <http://openenergy-platform.org/ontology/oeo-physical/ChemicalEnergy>
+Class: <http://openenergy-platform.org/ontology/oeo-physical/PortionOfCarbonDioxide>
+
+    Annotations: 
+        <http://purl.obolibrary.org/obo/IAO_0000115> "Carbon dioxide is a portion of matter with the chemical formula CO2. It is a gas that occurs naturally in the atmosphere as a trace gas and it is a greenhouse gas.",
```
```suggestion
        <http://purl.obolibrary.org/obo/IAO_0000115> "Carbon dioxide is a portion of matter with the chemical formula CO2. It is a gas that occurs naturally in the atmosphere as a trace gas and can work as a greenhouse gas.",
```

stap-m commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -269,6 +272,61 @@ Class: <http://openenergy-platform.org/ontology/oeo-physical/ChemicalEnergy>
+Class: <http://openenergy-platform.org/ontology/oeo-physical/PortionOfMethane>
+
+    Annotations: 
+        <http://purl.obolibrary.org/obo/IAO_0000115> "Methane is a portion of matter with the chemical formular CH4. It is a gas that occurs naturally in the atmosphere as a trace gas and it is a greenhouse gas. As it can be oxidised it is a fuel.",
```
```suggestion
        <http://purl.obolibrary.org/obo/IAO_0000115> "Methane is a portion of matter with the chemical formular CH4. It is a gas that occurs naturally in the atmosphere as a trace gas and can work as a greenhouse gas. As it can be oxidised it can be usede as a fuel.",
```

stap-m commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -269,6 +272,61 @@ Class: <http://openenergy-platform.org/ontology/oeo-physical/ChemicalEnergy>
+Class: <http://openenergy-platform.org/ontology/oeo-physical/PortionOfNitrogenTrifluoride>
+
+    Annotations: 
+        <http://purl.obolibrary.org/obo/IAO_0000115> "Nitrogen trifluoride is a portion of matter with the chemical formula NF3. It is a potent greenhouse gas from anthropogenic origin.",
```
```suggestion
        <http://purl.obolibrary.org/obo/IAO_0000115> "Nitrogen trifluoride is a portion of matter with the chemical formula NF3. It can work as a potent greenhouse gas and has an anthropogenic origin.",
```

stap-m commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -269,6 +272,61 @@ Class: <http://openenergy-platform.org/ontology/oeo-physical/ChemicalEnergy>
+Class: <http://openenergy-platform.org/ontology/oeo-physical/PortionOfNitrousOxide>
+
+    Annotations: 
+        <http://purl.obolibrary.org/obo/IAO_0000115> "Nitrous oxide is a portion of matter with the chemical formula N2O. It is a gas that occurs naturally in the atmosphere as a trace gas and it is a greenhouse gas.",
```
```suggestion
        <http://purl.obolibrary.org/obo/IAO_0000115> "Nitrous oxide is a portion of matter with the chemical formula N2O. It is a gas that occurs naturally in the atmosphere as a trace gas and it can work as a greenhouse gas.",
```

stap-m commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -269,6 +272,61 @@ Class: <http://openenergy-platform.org/ontology/oeo-physical/ChemicalEnergy>
+Class: <http://openenergy-platform.org/ontology/oeo-physical/PortionOfSulphurHexafluoride>
+
+    Annotations: 
+        <http://purl.obolibrary.org/obo/IAO_0000115> "Sulphur hexafluoride is a portion of matter with the chemical formula SF6. It is a potent greenhouse gas from anthropogenic origin.",
```
```suggestion
        <http://purl.obolibrary.org/obo/IAO_0000115> "Sulphur hexafluoride is a portion of matter with the chemical formula SF6. It can work as a potent greenhouse gas and has an anthropogenic origin.",
```

akleinau commented:
> Additionally to what I have commented on specific code parts two general things:
> 
> * 'has_origin anthropogenic' is missing for the fluorinated greenhouse gases.
> * The equivalent classes for CarbonDioxide, Methane and NitrousOxide as subclasses of GreenhouseGas and the equivalent classes for Hydrofluorocarbon, NitrogenTrifluoride, Perfluorocarbon and SulphurHexafluoride as subclasses of FluorinatedGreenhouseGas are missing. These are needed to keep the structure within greenhouse gases.

I added the origin but don't understand why the equivalent classes that were just Abbreviations are needed for the structure. I added them as alternative terms because they dont represent new concepts, just new words.

l-emele commented:
> 
> 
> > Additionally to what I have commented on specific code parts two general things:
> > 
> > * 'has_origin anthropogenic' is missing for the fluorinated greenhouse gases.
> > * The equivalent classes for CarbonDioxide, Methane and NitrousOxide as subclasses of GreenhouseGas and the equivalent classes for Hydrofluorocarbon, NitrogenTrifluoride, Perfluorocarbon and SulphurHexafluoride as subclasses of FluorinatedGreenhouseGas are missing. These are needed to keep the structure within greenhouse gases.
> 
> I added the origin but don't understand why the equivalent classes that were just abbreviations are needed for the structure. I added them as alternative terms because they dont represent new concepts, just new words.

I am not talking about the abbreviations but the classes with full names. If we move the greenhouse gases to PortionOfMatter we need the equivalents as subclasses of GreenhouseGas. Else we break again something existing.

akleinau commented:
Those gases are PortionsOfMatter with the property/ disposition GreenhouseGas, it would be logically false to classify those material entities as (immaterial) properties.
If we break sth by having moved them to PortionOfMatter that sth is also logically false and should be broken (but I dont see what that could be)

akleinau commented:
![image](https://user-images.githubusercontent.com/56925445/73923886-deb22800-48cb-11ea-8ebd-347dab6c0c15.png)
Please don't judge my art abilities. And that I visualised gas as a comic-style cloud. What I want to express is that we have "is a GreenhouseGas" currently as a property of a gas (like we have its origin as another example) and it would be false to take sth as material as a gas out of the material world and make it a property. Especially if we want to have it as sth material too.

jannahastings commented:
The disposition would be something like GreenhouseExacerbator

akleinau commented:
![image](https://user-images.githubusercontent.com/56925445/74660879-3fc4e000-5197-11ea-8fb0-fbb4dc383912.png)

We now have an equivalent class with all the GreenhouseGases and the disposition renamed to -Exacerbator. Please review again.

l-emele commented:
I am studying, working and modelling for about a decade now in the domain with a focus on greenhouse gases but I never heard of the terms _greenhouse exacerbator_ or _greenhouse gas exacerbator_. So I do not think that this is a useful term.

A common term though is the _greenhouse effect_. We have that already in the definition of greenhouse gases. Couldn't we derive something from that instead of inventing a completely new term?

jannahastings commented:
> I am studying, working and modelling for about a decade now in the domain with a focus on greenhouse gases but I never heard of the terms _greenhouse exacerbator_ or _greenhouse gas exacerbator_. So I do not think that this is a useful term.
> 
> A common term though is the _greenhouse effect_. We have that already in the definition of greenhouse gases. Couldn't we derive something from that instead of inventing a completely new term?

We could perhaps just call the relevant disposition GreenhouseEffect?

akleinau commented:
I associate with GreenhouseEffect a global phenomena and not a property of an object. This could lead to confusion

jannahastings commented:
How about GreenhouseEffectContributor?

l-emele commented:
> I associate with GreenhouseEffect a global phenomena and not a property of an object. This could lead to confusion

It is a global phenomenon but it is global because the greenhouse gases mix well with the main constituents of the atmosphere like nitrogen and oxygen.

It can be demonstrated that the effect is a property of the gas, e.g. in this television experiment from the 1970s: https://www.youtube.com/watch?v=lORAR1nvfjs (unfortunately only in German).

akleinau commented:
> > I associate with GreenhouseEffect a global phenomena and not a property of an object. This could lead to confusion
> 
> It is a global phenomenon but it is global because the greenhouse gases mix well with the main constituents of the atmosphere like nitrogen and oxygen.
> 
> It can be demonstrated that the effect is a property of the gas, e.g. in this television experiment from the 1970s: https://www.youtube.com/watch?v=lORAR1nvfjs (unfortunately only in German).

thanks that was a great video!

stap-m commented:
I'd be ok with _GreenhouseEffect_.

akleinau commented:
ok, I renamed to GreenhouseEffect and FlourinatedGreenhouseEffect (is that a correct term?)

stap-m commented:
> Else I'd delete that second effect and just leave the subclass FluorinatedGreenhouseGases?

Yes.

akleinau commented:
so, should I rather define fluorinated greenhouse gases as all greenhouse gases that are of anthropogenic origin or as an exact list of its four types?

l-emele commented:
Fluorinated greenhouse gases is that subset of all greenhouse gases which are produced by [fluorination](https://en.wikipedia.org/wiki/Fluorination). All fluorinated greenhouse gases are anthopogenic. The other greenhouse gases like carbon dioxide and methane _can_ be anthropogenic but can also be natural.

akleinau commented:
ok we dont have an origin called fluorination right now, should we add it?

l-emele commented:
Fluorination is not an origin but an process.

akleinau commented:
ok then I‘ll just name them directly for now

akleinau commented:
ok, next review please :)

akleinau commented:
ok after implementing the changes from the February ontohack as written in the protocol (with some changes bc it was a bit messy from many changes) the new structure is as follows:

continuants
- **PortionOfGreenhouseGas**:  PortionOfMatter that has the GreenhouseEffectDisposition.
- **GreenhouseEffectDisposition**:  disposition of a gas to contribute to GreenhousegasEmission.

processes
- **Emission**: process releasing byproducts from human activity (e.g. production, distribution or consumption) into the environment.
   - **GreenhousegasEmission**: Emission that releases a greenhouse gas.
   - **Pollution**: Emission with a negative effect on the environment or organisms.
- **NegativeEmission**: process of absorbing a substance (that was emitted before, typically a pollution).


notes
- I had thought about doing an own pull request for pollution but its all interconnected so I changed it here
- the disposition is mainly for defining the equivalent class and can alternatively be replaced by some relation of the gases to the process
- PortionOfGreenhouseGas will be renamed to GreenhouseGas when we delete the PortionOf prefixes

jannahastings commented:
This is great but I would consider changing this definition slightly: 

>     * **GreenhouseEffectDisposition**:  disposition of a gas to contribute to GreenhousegasEmission.

I think the greenhouse effect disposition is the disposition of a gas to contribute to the greehouse effect, when it is emitted into the atmosphere via an emission.

akleinau commented:
won't every GreenhousegasEmission contribute to the greenhouse effect?

jannahastings commented:
> won't every GreenhousegasEmission contribute to the greenhouse effect?

Yes but that still does not define what the nature of the disposition is. It's not a disposition to be emitted, it's a disposition to cause problems _of a certain type_ once it has been emitted.

stap-m commented:
I agree with @jannahastings. Also because we have definition in circles here: PortionOfGreenhouseGas refers to GreenhouseEffectDisposition refers to GreenhousegasEmission refers to greenhouse gas ... I guess this is not a good idea.
I'd propose to work on the def for GreenhouseEffectDisposition and to break the circle.

akleinau commented:
ok, I implemented the new definition

akleinau commented:
I used this request to try out the 'term tracker item' annotation to add: "issue #56, pull request #234" to the upper classes.
What do you think of this notation, anything missing, or is just the issue enough?

jannahastings commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -1170,37 +1243,34 @@ Class: Graph
     Annotations: 
-        <http://purl.obolibrary.org/obo/IAO_0000115> "A greenhouse gas is a gas that absorbs and emits radiant energy within the thermal infrared range. Greenhouse gases cause the greenhouse effect.",
-        <http://purl.obolibrary.org/obo/IAO_0000119> "https://en.wikipedia.org/w/index.php?title=Greenhouse_gas&oldid=880592155"
+        <http://purl.obolibrary.org/obo/IAO_0000115> "The greenhouse effect disposition is the disposition of a gas to contribute to the greehouse effect, when it is emitted into the atmosphere via an emission.",
```
Just a small typo: 'greehouse effect' should be 'greenhouse effect'

jannahastings commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -1607,6 +1676,16 @@ Class: NaturalGasGenerator
+Class: NegativeEmission
+
+    Annotations: 
+        <http://purl.obolibrary.org/obo/IAO_0000115> "NegativeEmission is the process of absorbing a substance (that was emitted before, typically a pollution).",
```
Having a part of the definition text enclosed in brackets makes it seem a bit uncommitted. Also I don't think it is possible to say "a pollution". How about: 
"NegativeEmission is the process of absorbing a substance, usually a pollutant that was emitted before."

akleinau commented:
@jannahastings ok, changes made

stap-m commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -1312,17 +1382,16 @@ Class: Hydroelectricity
     
     SubClassOf: 
-        FluorinatedGreenhouseGas
+        EnergyStorage,
```
```suggestion
        PortionOfMatter
```

stap-m commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -1789,7 +1854,7 @@ Class: ParticulateMatter
     
     SubClassOf: 
-        AirPollutant
+        AirPollution
```
```suggestion
        PortionOfMatter
```
AirPollution is a process/emission, ParticulateMatter is a portion of matter which is emitted.

stap-m commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -2898,7 +2977,7 @@ Some VOCs are dangerous to human health or cause harm to the environment."@en,
     
     SubClassOf: 
-        AirPollutant
+        AirPollution
```
```suggestion
        PortionOfMatter
```
same as particulateMatter

akleinau commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -1170,37 +1243,34 @@ Class: Graph
-        <http://purl.obolibrary.org/obo/IAO_0000115> "A greenhouse gas is a gas that absorbs and emits radiant energy within the thermal infrared range. Greenhouse gases cause the greenhouse effect.",
-        <http://purl.obolibrary.org/obo/IAO_0000119> "https://en.wikipedia.org/w/index.php?title=Greenhouse_gas&oldid=880592155"
+        <http://purl.obolibrary.org/obo/IAO_0000115> "The greenhouse effect disposition is the disposition of a gas to contribute to the greenhouse effect, when it is emitted into the atmosphere via an emission.",
+        <http://purl.obolibrary.org/obo/IAO_0000233> "issue #56, pull request #234"
```
good question, I'll address this in the #259 issue

l-emele replied:
I'd prefer links as you can click on them.

akleinau replied:
ok I guess links do good and no harm so theirs no reason not to include them

l-emele commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -309,6 +315,88 @@ Class: <http://openenergy-platform.org/ontology/oeo-physical/NuclearFuel>
+Class: <http://openenergy-platform.org/ontology/oeo-physical/PortionOfFluorinatedGreenhouseGas>
+
+    Annotations: 
+        <http://purl.obolibrary.org/obo/IAO_0000115> "A PortionOfFluorinatedGreenhouseGas is a PortionOfGreenhouseGas that is of anthropogenic origin."
```
The current definition implies that all anthropogenic greenhouse gases are fluorinated but that is in reality not the case.
```suggestion
        <http://purl.obolibrary.org/obo/IAO_0000115> "A PortionOfFluorinatedGreenhouseGas is a PortionOfGreenhouseGas that is produced by fluorination. Hence they contains Fluor (F) atoms and are of anthropogenic origin."
```
I am aware that neither the terms _fluorination_ nor _fluor_ are part of the OEO and I don't think that it is necessary to include these. If we want to include these, we might link to ChEBI.

l-emele commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -581,7 +669,7 @@ Class: CarbonDioxideEmissionTrafficSector
     
     SubClassOf: 
-        CarbonDioxideEmission
+        Sector
```
`CarbonDioxideEmissionTrafficSector` is not a sector, but I am fine to shift this class **temporarily** to sector and deal with it in issue #170.

l-emele commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -1170,37 +1243,34 @@ Class: Graph
     
     
-Class: Grid
+Class: GreenhousegasEmission
```
If we include the class `GreenhousegasEmission` we should make `CarbonDioxideEmission` a subclass of it. Currently it is also a direct subclass of `Emission`.

akleinau replied:
changed it

l-emele commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -1846,14 +1898,14 @@ Class: PlantOperation
     
     SubClassOf: 
-        <http://purl.obolibrary.org/obo/BFO_0000016>
+        Emission
```
I don't think, a subclass relationship is correct here. The emission of a pollutant is the cause for a pollution.

Example: If you spill mud into a pool of clean water the water gets dirty. The mud in the water is the _pollution_ but _emission_ is the process of spilling the mud into the water.

akleinau replied:
The def of Pollution: "Pollution is an Emission with a negative effect on the environment or organisms." refers to the process of polluting. So do we need both the process and the object or is one enough?

akleinau replied:
I think pollution will cause some more discussion, maybe we leave it as it is for this request and discuss it more thoroughly in #263?

l-emele replied:
You might be right, probably I misunderstand. What is then the difference between an emission and a pollution?

akleinau replied:
pollution has a negative impact, emission just an impact

l-emele replied:
Ah, sorry... I read over the little word _negative_. So it is fine to leave it as it is.

l-emele commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -2593,6 +2669,9 @@ Class: Sea
         GeographicRegion
     
     
+Class: Sector
```
Here a class `Sector` pops up without any definition or any relation to other classes. I think this is caused by changing `CarbonDioxideEmissionTrafficSector` to subclass of `Sector`. The class `Sector` is defined in the oeo-social. Does the import work not properly?

akleinau replied:
oh yes I moved it manually now from physical to social
